applications/asset_tracker_v2/src/modules/location_module.c

```diff
@@ -231,17 +231,23 @@ static void search_start(void)
 {
 	int err;
 	struct location_config config;
-	enum location_method methods[2];
+	enum location_method methods[3];
 	int methods_count = 0;
 	int methods_index_gnss = -1;
 	int methods_index_cellular = -1;
 
-	if (copy_cfg.no_data.neighbor_cell && copy_cfg.no_data.gnss) {
+	if (copy_cfg.no_data.neighbor_cell && copy_cfg.no_data.gnss && copy_cfg.no_data.wifi) {
 		SEND_EVENT(location, LOCATION_MODULE_EVT_DATA_NOT_READY);
 		LOG_ERR("Both GNSS and cellular are configured off");
 		return;
 	}
 
+#if defined(CONFIG_LOCATION_METHOD_WIFI)
+	if (!copy_cfg.no_data.wifi) {
+		methods[methods_count] = LOCATION_METHOD_WIFI;
+		methods_count++;
+	}
+#endif
```

Comment:
Wi-Fi is now set as 1st priority method. Should it be after GNSS?
The reason it might be fine here is that ATv2 doesn't care about the result of the Wi-Fi positioning so GNSS is started anyway.